app/controllers/admin/sessions_controller.rb

```diff
@@ -5,10 +5,74 @@ class SessionsController < Devise::SessionsController
     before_action :user_logout_required, except: :destroy
     skip_before_action :store_location, raise: false
 
+    prepend_before_action :authenticate_with_otp_two_factor, 
+                          if: -> { action_name == "create" && otp_two_factor_enabled? }
+    
+    protect_from_forgery with: :exception, prepend: true, except: :destroy
+
     # GET /admin/logout
     def confirm_logout
       # If the user is already logged out, we just redirect to the front page.
       redirect_to root_path unless admin_signed_in?
     end
+
+    # Two-Factor Authentication
+    def authenticate_with_otp_two_factor
```

Comment:
Is all this required, or can the default TOTP strategy with Devise be used? I ask because of this line in the devise-two-factor docs:
> These parameters can be submitted to the standard Devise login route, and the strategy will handle the authentication of the user for you.

(https://github.com/devise-two-factor/devise-two-factor?tab=readme-ov-file#logging-in)

Reply:
It isn't if you don't want the two-step login behaviour. I can turn it back into a one-step form, then pretty much all the new code in this file can be deleted. But the TOTP field would be there no matter if TOTP is enabled or not

Reply:
Darn, that's unfortunate. Thanks for the explanation!